### ISSUE TITLE
`Hotwire.registerBridgeComponents()` doesn't set user agent

### ISSUE THREAD

**Issue by joemasilotti** [hotwired/hotwire-native-android#68]
The documentation on bridge components states the following is all that's needed to register your components:

```kotlin
Hotwire.registerBridgeComponents(
    BridgeComponentFactory("button", ::ButtonComponent)
)
```

However, this will *not* add the component names to the user agent.

It seems that calling `Hotwire.config.userAgentSubstring()` is needed to gather the registered components correctly. Interestingly enough, `Hotwire.config.userAgent` does *not* seem to be enough.

You can replicate this by commenting out `DemoApplication.kt:72` in the demo app and noticing that no bridge components are rendered.

---

I could update the documentation to reflect this required addition. But I'd rather it work like iOS - registering bridge components sets the user agent automatically.

**Comment by EmilioCristalli** [comment]
this might be related to this issue https://github.com/hotwired/hotwire-native-android/issues/64

**Comment by martron** [comment]
This issue sent me down a debugging rabbit hole for a couple hours trying to get my first BridgeComponent working.  It was unintuitive to realize I need to set a custom user-agent in order to get the BridgeComponent stimulus controllers to fire.

**Comment by jayohms** [comment]
This was resolved in https://github.com/hotwired/hotwire-native-android/pull/77 and will be available in `v1.1` soon.
